Proofs/Experiments/Abs/AbsVCGTandem.lean

```diff
@@ -253,42 +276,35 @@ theorem partial_correctness :
     replace h_s3_run := h_s3_run.symm
     sym_n 1 at s3
     have h_s4_cut := @program.stepi_0x4005dc_cut s3 s4
-                      h_s3_program h_s3_pc h_s3_err h_s3_sp_aligned
+                      h_s3_program h_s3_pc h_s3_err
                       (by simp only [run, stepi_s3])
     -- Note: from the conclusion, we see that we simulated 3 steps from s1 (or 4
     -- steps from s0) to reach the next cutpoint. We can use this to help
     -- us come up with a clock in the termination proof.
     simp only [h_s4_cut, Nat.reduceAdd, minimal_theory]
     clear h_s3_run h_s4_cut stepi_s3
 
-    -- End: Symbolic Simulation
-
     simp only [abs_assert, abs_post, h_pre, minimal_theory]
     -- Aggregate program effects here to obtain the value of x0(s4).
     sym_aggregate
+    -- End: Symbolic Simulation
 
     simp (config := {ground := true}) only [AddWithCarry, spec]
-    split <;> bv_decide
+    constructor
+    · -- Partial Correctness Proof
+      split <;> bv_decide
+    · -- Termination Proof
+      apply Exists.intro 4
+      simp only [Spec'.cut, abs_cut, rank,
+                 state_simp_rules]
+      have h_run : run 4 s0 = s4 := by
+        -- (TODO @alex) If we kept track of the number of steps we took during
```

Comment:
@alexkeizer Here's a request for you!

Reply:
We do track the steps, thats why I added the `stepi s{i} = s{i+1}` local hypotheses!

At first I thought you wanted to keep `h_run` around with its original type, but we'd have to aggregate here anyway, because we only simulate one step at a time, so it'd be effectively the same as just using the `stepi_s{j}` lemmas

Reply:
Thanks! I'd accidentally cleared those `stepi_` hyps and then forgotten all about them!